File: examples/typescript/testing/assertions/assert-step.ts
import { withDurableExecution, DurableContext } from "@aws/durable-execution-sdk-js";
import { LocalDurableTestRunner, OperationStatus } from "@aws/durable-execution-sdk-js-testing";
import { ExecutionStatus, OperationType } from "@aws-sdk/client-lambda";

const handler = withDurableExecution(async (event: unknown, context: DurableContext) => {
  return await context.step("compute", () => 42);
});

let runner: LocalDurableTestRunner;

beforeAll(async () => {
  await LocalDurableTestRunner.setupTestEnvironment();
});

afterAll(async () => {
  await LocalDurableTestRunner.teardownTestEnvironment();
});

beforeEach(() => {
  runner = new LocalDurableTestRunner({ handlerFunction: handler });
});

it("asserts on a step operation", async () => {
  const result = await runner.run();

  expect(result.getStatus()).toBe(ExecutionStatus.SUCCEEDED);

  const step = runner.getOperation("compute");
  expect(step.getType()).toBe(OperationType.STEP);
  expect(step.getStatus()).toBe(OperationStatus.SUCCEEDED);
  expect(step.getStepDetails()?.result).toBe(42);
});

File: examples/java/testing/assertions/assert-step.java
import static org.junit.jupiter.api.Assertions.*;

import org.junit.jupiter.api.Test;
import software.amazon.awssdk.services.lambda.model.OperationStatus;
import software.amazon.awssdk.services.lambda.model.OperationType;
import software.amazon.lambda.durable.DurableContext;
import software.amazon.lambda.durable.model.ExecutionStatus;
import software.amazon.lambda.durable.testing.LocalDurableTestRunner;

class AssertStepTest {

    @Test
    void assertsOnStepOperation() {
        var runner = LocalDurableTestRunner.create(
            Void.class,
            (input, context) -> context.step("compute", Integer.class, ctx -> 42)
        );

        var result = runner.runUntilComplete(null);

        assertEquals(ExecutionStatus.SUCCEEDED, result.getStatus());

        var step = result.getOperation("compute");
        assertNotNull(step);
        assertEquals(OperationType.STEP, step.getType());
        assertEquals(OperationStatus.SUCCEEDED, step.getStatus());
        assertEquals(42, step.getStepResult(Integer.class));
    }
}
